packages/cli/src/start/server.ts

```diff
@@ -0,0 +1,199 @@
+import fs from 'node:fs'
+import path from 'node:path'
+
+import { type AnyModels, handleGetRequest } from '@rise-tools/server'
+import chokidar from 'chokidar'
+import Fastify from 'fastify'
+import internalIp from 'internal-ip'
+import qrcode from 'qrcode-terminal'
+import { WebSocketServer } from 'ws'
+import z from 'zod'
+
+// TODO
+import { connectWebSocket, createWSServerContext } from '../../../server/src/ws-connection'
+import { IGNORED_PATH, WATCH_PATH } from '../config/constants'
+
+interface IndexedModels {
+  [key: string]: any
+}
+
+export const devArgsSchema = z.object({
+  ws: z.boolean().optional().default(false),
+  prod: z.boolean().optional().default(false),
+  port: z.number(),
+  host: z.string(),
+  cwd: z.string().optional(),
+})
+
+export type DevArgs = z.infer<typeof devArgsSchema>
+
+class ModelManager {
+  models: AnyModels & IndexedModels = {}
+  cwd: string = process.cwd()
+
+  constructor(cwd: string) {
+    this.cwd = cwd
+  }
+
+  parseModelPath(modelPath: string) {
+    const key = modelPath.split('/').slice(1, -1).join(':')
+    const path = modelPath.split('.').slice(0, -1).join('.')
+    return { key, path }
+  }
+
+  async updateModel(modelPath: string) {
+    const { key, path: basePath } = this.parseModelPath(modelPath)
+    const cwd = this.cwd!
+    this.models[key] = await import(path.join(cwd, basePath))
+  }
+
+  removeModel(modelPath: string) {
+    const { key } = this.parseModelPath(modelPath)
+    this.models[key]
+  }
+
+  getModels() {
+    return this.models
+  }
+}
+
+export class DevServer {
+  options: DevArgs
+  modelManager: ModelManager
+
+  constructor(options: DevArgs) {
+    this.options = options
+    this.modelManager = new ModelManager(options.cwd!)
+
+    this.start()
+  }
+
+  async start() {
+    const port = this.options.port
+    const host = this.getHost()
+
+    const fsWatcher = await this.watchFiles()
+
+    if (this.options.prod) {
+      // Close the watcher if we are in prod mode after adding all models
+      fsWatcher.close()
+    }
+
+    this.startServer()
+
+    console.log('Server started on', `${host}:${port}`)
+    console.log('Scan the QR from rise playground')
+
+    this.createDevQR(`rise-playground://${host}:${port}`)
```

Comment:
nit: I am sure Eric will know better (my memory is fading away), but I think our link protocol is just `rise://` 🙏

Reply:
yes the protocol is just `rise:` and also the URL format is significantly different. See the code here to understand how rise URLs are actually created: 

https://github.com/rise-tools/rise-tools/blob/main/apps/mobile/src/screens/qr-code.tsx#L19-L20

basically we take a "Connection" object with several fields, json stringify it, base 58 encode it, then pass as a query param to `rise://connect?connectInfo=...`